lib/helpers/generateFilterType.ts

```diff
@@ -24,7 +24,8 @@ export const createFilterType = (type: Function) => {
   const typeGraphQLMetadata = getTypeGraphQLMetadataStorage();
 
   const objectTypesList = typeGraphQLMetadata.objectTypes;
-  const graphQLModel = objectTypesList.find((ot) => types.includes(ot.target) && (typeGraphQLMetadata.inputTypes.findIndex(it => it.name === (ot.name + "Filter")) < 0))
+  const graphQLModels = objectTypesList.filter((ot) => types.includes(ot.target) && (typeGraphQLMetadata.inputTypes.findIndex(it => it.name === (ot.name + "Filter")) < 0));
+  const graphQLModel = graphQLModels.find(m => m.name === type.name) || graphQLModels.slice(-1).pop()
```

Comment:
Although, it's not critical in this particular case, but it would be better to use `graphQLModels[graphQLModels.length - 1]` to get last value of array instead of `graphQLModels.slice(-1).pop()` (to avoid redundant copying of an array)